### PR TITLE
now wrapping "try excet" statement with ipdbugger as well

### PR DESCRIPTION
**Pull request by yakobu**

### CODE REVIEW

File: ipdbugger/__init__.py [hunk 1]
```diff
@@ -241,9 +273,6 @@ def wrapper(*args, **kw):
 
                 tree.body[0].body.insert(1, import_exception_cmd)
 
-            # Delete the debugger decorator of the function
```
Comment (UnDarkle) [inline ipdbugger/__init__.py]: we needed it (for some reason)

File: ipdbugger/__init__.py [hunk 2]
```diff
@@ -139,6 +140,37 @@ def __init__(self, ignore_exceptions=(), catch_exception=None):
...
+        items = [self.visit(item) for item in node.body]
+        node.body = items
+
+        # Remove from ignore list
+        for except_handler in excepted:
+            self.exception_handlers.remove(except_handler)
+
+        return node
+
+    if sys.version_info > (3, 0):  # pragma: no cover
```
Comment (UnDarkle) [inline ipdbugger/__init__.py]: make sure we also wrap the try-except statement itself with ipdbugger

**Review by UnDarkle — CHANGES_REQUESTED** [review]
2 comments

File: ipdbugger/__init__.py [hunk 3]
```diff
@@ -139,6 +141,33 @@ def __init__(self, ignore_exceptions=(), catch_exception=None):
...
+    def try_except_handler(self, node):
+        """Handler for try except statement to ignore excepted exceptions."""
+        # List all excepted handlers
+        excepted = [ast.ExceptHandler(type=handler.type,
+                                      name=None,
+                                      body=[ast.Raise()])
+                    for handler in node.handlers]
+
+        new_exception_handlers = []
+        for except_handler in excepted + self.exception_handlers:
```
Comment (osherdp) [inline ipdbugger/__init__.py]: use concat* from itertools

File: ipdbugger/__init__.py [hunk 4]
```diff
@@ -148,27 +177,35 @@ def generic_visit(self, node):
...
-                    finalbody=[],
-                    handlers=self.exception_handlers)
+            is_python_3 = sys.version_info > (3, 0)
+            ast_try_except = ast.Try if is_python_3 else ast.TryExcept
+            try_except_extra_params = {"finalbody": []} if is_python_3 else {}
+
+            new_node = ast_try_except(
+                orelse=[],
+                body=[node],
+                handlers=self.exception_handlers[:],
```
Comment (osherdp) [inline ipdbugger/__init__.py]: check if you still need to clone, since you're creating a new list inside

File: ipdbugger/__init__.py [hunk 5]
```diff
@@ -148,27 +177,35 @@ def generic_visit(self, node):
...
         if (isinstance(node, ast.stmt) and
                 not isinstance(node, ast.FunctionDef)):
 
-            if sys.version_info > (3, 0):  # pragma: no cover
-                new_node = ast.Try(  # pylint: disable=no-member
-                    orelse=[],
-                    body=[node],
-                    finalbody=[],
-                    handlers=self.exception_handlers)
+            is_python_3 = sys.version_info > (3, 0)
```
Comment (osherdp) [inline ipdbugger/__init__.py:184]: elkis, whady'a think?

File: setup.py [hunk 6]
```diff
@@ -1,7 +1,7 @@
 """Setup file for handling packaging and distribution."""
 from setuptools import setup
 
-__version__ = "2.0.3"
+__version__ = "2.1.3"
```
Comment (osherdp) [inline setup.py]: 2.1.0

File: tests/test_debug.py [hunk 7]
```diff
@@ -136,3 +136,59 @@ def func():
...
+            raise ValueError()
+        except ValueError:
+            raise
+
+    with capture_output(), patch('bdb.Bdb.set_trace') as set_trace:
+        func()
+        assert set_trace.called_once
+
+
+def test_wrapping_try_except_statement_with_no_specific_exception_type():
```
Comment (osherdp) [inline tests/test_debug.py]: test_wrapping_general_try_except, or unbound catch

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ipdbugger/__init__.py (hunk 1 above)
Comment (yakobu) [reply ipdbugger/__init__.py]: K

File: ipdbugger/__init__.py (hunk 3 above)
Comment (yakobu) [reply ipdbugger/__init__.py]: K

**Pull request merged**